src/cn-batch-forms.service.js

```diff
@@ -97,6 +97,29 @@ export function setValue(ffService) {
           originalVal;
         update.set(updateVal);
       }
+      else if (_.isObject(originalVal) && _.isObject(val)) {
+        let newVal = _.cloneDeep(originalVal)
+        for (let key in val) {
+          if (key in originalVal && _.isArray(val[key]) && _.isArray(originalVal[key])) {
+            let found = false;
+            for (let arrayVal of val[key]) {
+              found = false;
+              for (let ogArrayVal of originalVal[key]) {
+                if (_.isEqual(ogArrayVal, arrayVal)) {
+                  found = true;
+                  break;
+                }
+                if (!found) {
+                  newVal[key].push(arrayVal);
+                }
+              }
+            }
+          } else {
+            newVal[key] = val[key];
+          }
+        }
+        update.set(newVal);
```

Comment:
Or if you have to make sure that every value in each is also an array, you might do something like:

```js
const filter = _.partialRight(_.filter, _.isArray);
update.set(_.intersection(filter(val), filter(originalVal)));
```

The `partialRight` function is because Lodash's function arguments are backwards, so we have to flip them around in order to compose them ([Lodash FP](https://github.com/lodash/lodash/wiki/FP-Guide) fixes that).

Reply:
Or I guess the `deepClone` part might be important, in that case something like this might work (note I haven't tested any of these suggestions):

```js
let val_ =
  _(val)
  .deepClone()
  .filter(_.isArray)
  .value();
let originalVal_ =
  _(originalVal)
  .deepClone()
  .filter(_.isArray)
  .value();
update.set(_.intersection(val_, originalVal_);
```

Reply:
I'm not fully sure to understand either. Maybe if I give you an example you will understand better.
So the var `val` can be set to:
```
{
  a: [{
    id: 1,
    name: 'name 1'
 }],
 b: [1,  2]
}
```
and the updated value can be:
```
{
  a: [{
    id: 2,
    name: 'name 2'
 }],
 b: [3],
 c: [{
   id: 123,
   path: 'travel > yolo'
 }]
}
```

And this should give this updated object:
```
{
  a: [{
    id: 1,
    name: 'name 1'
 }, {
    id: 2,
    name: 'name 2'
 }],
 b: [1, 2, 3],
 c: [{
   id: 123,
   path: 'travel > yolo'
 }]
}
```
Would your solution make the code cleaner?

Reply:
Ah yeah the example helps! I definitely misunderstood the code. It seems what you want is:

```js
var newVal = _.merge(_.cloneDeep(val), _.cloneDeep(originalVal), (a, b) =>
   _.isArray(a) ? a.concat(b) : a
);
```

You can test the behavior in the console: 
![Screen Shot 2021-05-04 at 12 51 10 PM](https://user-images.githubusercontent.com/709141/117062014-bb954480-acd7-11eb-852e-c2052a6dbf5e.png)

Reply:
@davezuch  I ended up using your suggestion plus `Set` to make sure the values are unique.
```
_.merge(_.cloneDeep(originalVal), _.cloneDeep(val), (a, b) =>
           _.isArray(a) ? [...new Set(a.concat(b))] : a
```